plugins/modules/inventory_workflow_manager.py

```diff
@@ -2612,16 +2628,23 @@ def update_interface_detail_of_device(self, device_to_update):
                     self.log("Received API response from 'update_interface_details': {0}".format(str(response)), "DEBUG")
 
                     if response and isinstance(response, dict):
-                        task_id = response.get('response').get('taskId')
+                        response = response.get('response')
+                        if not response:
+                            self.status = "failed"
+                            self.msg = "Interface Updation get failed due to empty response from the update_interface_details API."
+                            self.log(self.msg, "ERROR")
+                            continue
+
+                        task_id = response.get('taskId')
 
                         while True:
                             execution_details = self.get_task_details(task_id)
 
                             if 'SUCCESS' in execution_details.get("progress"):
                                 self.status = "success"
-                                self.result['changed'] = True
-                                self.result['response'] = execution_details
+                                is_update_occurred = True
                                 self.msg = "Updated Interface Details for device '{0}' successfully".format(device_ip)
```

Comment:
self.msg = "Successfully updated the Interface Details for device '{0}'.".format(device_ip)

Reply:
updated.